components/esp_modem/test/target_ota/components/manual_ota/transport_batch_tls.cpp

```diff
@@ -0,0 +1,232 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Unlicense OR CC0-1.0
+ */
+#include <vector>
+#include "esp_log.h"
+#include "mbedtls_wrap.hpp"
+#include "esp_transport_tcp.h"
+
+#define TAG "batch-tls"
+
+class TlsTransport: public Tls {
+public:
+    explicit TlsTransport(esp_transport_handle_t parent) : Tls(), transport_(parent), read_len(0), offset(0) {}
+    int send(const unsigned char *buf, size_t len) override;
+    int recv(unsigned char *buf, size_t len) override;
+    static bool set_func(esp_transport_handle_t tls_transport);
+    int preread(size_t len, int timeout_ms);
+    bool prepare_buffer(size_t max_size);
+private:
+    esp_transport_handle_t transport_{};
+    int connect(const char *host, int port, int timeout_ms);
+    void delay() override;
+
+    struct transport {
+        static int connect(esp_transport_handle_t t, const char *host, int port, int timeout_ms);
+        static int read(esp_transport_handle_t t, char *buffer, int len, int timeout_ms);
+        static int write(esp_transport_handle_t t, const char *buffer, int len, int timeout_ms);
+        static int close(esp_transport_handle_t t);
+        static int poll_read(esp_transport_handle_t t, int timeout_ms);
+        static int poll_write(esp_transport_handle_t t, int timeout_ms);
+        static int destroy(esp_transport_handle_t t);
+    };
+    int last_timeout;
+    std::vector<char> buf;
+    size_t read_len;
+    size_t offset;
+};
+
+esp_transport_handle_t esp_transport_tls_init(esp_transport_handle_t parent)
```

Comment:
It seems to me that this could be simplified if we just change the ssl functions and add the context data. The relation with ` TlsTransport` and `ssl` is a bit confusing.

Reply:
We need to use tcp-transport handle from ESP-IDF.
Suggest renaming `ssl` and `tls` to something more descriptive.
* `ssl` -> `transport_handle`
* `tls` -> `tls_context`